src/tests/rules/valid-scripts.test.ts

```diff
@@ -0,0 +1,61 @@
+import { rule } from "../../rules/valid-scripts.js";
+import { ruleTester } from "./ruleTester.js";
+
+ruleTester.run("valid-scripts", rule, {
+	invalid: [
+		{
+			code: `{ "scripts": null }`,
+			errors: [{ messageId: "incorrectTypeScriptsField" }],
+		},
+		{
+			code: `{ "scripts": 123 }`,
+			errors: [{ messageId: "incorrectTypeScriptsField" }],
+		},
+		{
+			code: `{ "scripts": "" }`,
+			errors: [{ messageId: "incorrectTypeScriptsField" }],
+		},
+		{
+			code: `{ "scripts": [] }`,
+			errors: [{ messageId: "incorrectTypeScriptsField" }],
+		},
+		{
+			code: `{ "scripts": "{}" }`,
+			errors: [{ messageId: "incorrectTypeScriptsField" }],
+		},
+		{
+			code: `{ "scripts": {} }`,
+			errors: [{ messageId: "emptyScriptsField" }],
+		},
+		{
+			code: `{ "scripts": {
+  "test": ""
+} }`,
+			errors: [{ messageId: "emptyScript" }],
+		},
+		{
+			code: `{ "scripts": { "test": null }}`,
+			errors: [{ messageId: "incorrectTypeScript" }],
+		},
+		{
+			code: `{ "scripts": { "test": undefined }}`,
+			errors: [{ messageId: "incorrectTypeScript" }],
+		},
```

Comment:
[Testing] No need, `undefined` isn't valid JSON.

```suggestion
```

I'm actually surprised this doesn't cause any _"hey you have invalid stuff"_ complaint in the tester...